src/print.rs

```diff
@@ -16,6 +16,44 @@ use crate::module::ALL_MODULES;
 use crate::modules;
 use crate::segment::Segment;
 
+pub struct Grapheme<'a>(&'a str);
+
+pub trait UnicodeWidthGraphemes {
+    fn width_graphemes(&self, is_extended: bool) -> usize;
+}
+
+// Assume that graphemes have width of the first character in the grapheme
+impl<'a> UnicodeWidthGraphemes for Grapheme<'a> {
+    #[inline]
+    fn width_graphemes(&self, _is_extended: bool) -> usize {
+        self.0.chars().next().and_then(|c| c.width()).unwrap_or(0)
```

Comment:
I think might `max` be slightly more correct:
```suggestion
        self.0.chars().max(|c| c.width()).unwrap_or(0)
```